src/inc/testf.h

```diff
@@ -89,31 +89,24 @@ struct bao_test {
             testframework_fails);                                           \
     } while (0)
 
-#define BAO_TEST(suite, test)                                         \
-    static void func_bao_test_##suite##_##test(unsigned char*);       \
-    static void func_bao_test_TBD_##suite##_##test(void)              \
-    {                                                                 \
-        extern unsigned int testframework_tests;                      \
-        extern unsigned int testframework_fails;                      \
-        unsigned char failures = 0;                                   \
-        if (TESTF_LOG_LEVEL > 1) {                                    \
-            BAO_INFO_TAG();                                           \
-            printf("Running " #suite "\t" #test "\n");                \
-        }                                                             \
-        testframework_tests++;                                        \
-        func_bao_test_##suite##_##test(&failures);                    \
-        if (failures)                                                 \
-            testframework_fails++;                                    \
-    }                                                                 \
-    struct bao_test struct_bao_test_##suite##_##test __attribute__((  \
-        section(".testframework." #suite))) = { .suite_name = #suite, \
-        .test_name = #test,                                           \
-        .func_ptr = func_bao_test_TBD_##suite##_##test };             \
-    static void func_bao_test_##suite##_##test(unsigned char* failures)
-
-static inline void run_all();
-static inline int run_specific_test(char* suite, char* test);
-static inline int run_suite(char* suite);
-void bao_test_entry(void);
-
-#endif // BAO_TEST_H
+#define BAO_TEST(suite, test)                          \

```

Comment:
I agree with you. It doesn't read well.

What if I remove the TAG completely? And leave just TEST(suite, teste)? It would be generic enough to accommodate future changes to what we call "BAO".